### ISSUE TITLE
Historic variables API crashes after terminating process instance during multi-instance sub process or call activity

### ISSUE THREAD

**Issue by verheyenkoen** [flowable/flowable-engine#3536]
**Describe the bug**
Consulting the `GET /history/historic-variable-instances` API crashes for variables "nrOfActiveInstances" and "nrOfCompletedInstances" variables when you (prematurely) stop a process instance (with the `DELETE runtime/process-instances/{processInstanceId}` API) during a multi-instance execution (eg. using the `subProcess` or `callActivity` constructs).

**Expected behavior**
Loading the historic variables should never crash obviously.

**Code**
Deploy this model ([ModelWithSubProcess.bpmn20.xml.zip](https://github.com/flowable/flowable-engine/files/10114350/ModelWithSubProcess.bpmn20.xml.zip)) to a Flowable engine, start a new process instance using the `POST runtime/process-instances` API, retrieve the process instance id from the response body and stop it again using the `DELETE runtime/process-instances/{processInstanceId}`

Then go and retrieve the historic variables using the `GET history/historic-variable-instances?processInstanceId={processInstanceId}&size=500` API.

This will result in an HTTP 500 server error with content: 
<img width="327" alt="image" src="https://user-images.githubusercontent.com/599538/204570166-958a29a5-a04f-4056-9752-79735b6ec847.png">

... and log the following exception: 
[flowable-exception.txt](https://github.com/flowable/flowable-engine/files/10114434/flowable-exception.txt)

Notice that this does work if you query for a specific variableName as long as it is not "nrOfActiveInstances" or "nrOfCompletedInstances".

If you look at the database, the most noticeable thing is that the `var_type_` in `act_hi_varinst` for both variables is "bpmnParallelMultiInstanceCompleted". This is also the case when the process is still running, but at that time the API does not "crash". If you complete the process instance in a normal manner (by submitting the 3 tasks until an end event is reached), the `var_type_` for both variables is changed to "integer" and the API also continues to work as expected.

**Additional context**
Tested in Flowable 6.7.2
Able to reproduce in our own production and UAT environments as well as in an empty spawned [docker flowable-ui container from docker hub](https://hub.docker.com/r/flowable/flowable-ui).

The same bug occurred when we use a multi-instance `callActivity` (as opposed to a mutli-instance `subProcess` in the example model above).

**Comment by filiphr** [comment]
Thanks for the detailed issue report @verheyenkoen. We are going to look into this and try to fix the problem.

**Comment by filiphr** [comment]
This will be fixed with #2540. In any case the problem on main is slightly different (there is no exception, but 0 is returned) due to a fix we did for another bug in https://github.com/flowable/flowable-engine/commit/d2236c4025a079e8ac0fd97845728f67bbaf88ac

**Comment by filiphr** [comment]
Yes you are right @verheyenkoen. I've updated my comment from before. Thanks for pointing it out

**Comment by Yojay97** [comment]
Thank you very much
